File: data/readable_format/fortran/file2357.f90
program main
    implicit none
    real, dimension(10) :: vector
    real, dimension(5)  :: subvector
    integer :: i

    ! Initialize the vector with some values
    do i = 1, 10
        vector(i) = real(i)
    end do

    ! Assign every other element to subvector
    subvector = vector(1:10:2)

    ! Print subvector to verify
    print *, 'Subvector: ', subvector

    ! Verify the correctness
    call verify_subvector(subvector)
end program main

subroutine verify_subvector(subvector)
    implicit none
    real, dimension(5), intent(in) :: subvector
    real, dimension(5) :: expected
    integer :: i

    ! Expected values in the subvector
    expected = (/ 1.0, 3.0, 5.0, 7.0, 9.0 /)

    ! Check each element
    do i = 1, 5
        if (subvector(i) /= expected(i)) then
            print *, 'Test failed: subvector(', i, ') = ', subvector(i), ' expected ', expected(i)
            stop
        end if
    end do

    print *, 'Test passed: subvector is correct'
end subroutine verify_subvector

File: data/readable_format/cpp/file2357.cpp
#include <iostream>
#include <vector>

int main() {
    std::vector<float> vector(10);
    std::vector<float> subvector(5);

    // Initialize the vector with some values (optional, for demonstration)
    for (int i = 0; i < 10; ++i) {
        vector[i] = static_cast<float>(i + 1);  // Example initialization
    }

    // Assign every other element from vector to subvector
    for (int i = 0; i < 5; ++i) {
        subvector[i] = vector[i * 2];
    }

    // Print the subvectors to verify the result (optional)
    std::cout << "Subvector: ";
    for (const auto &val : subvector) {
        std::cout << val << " ";
    }
    std::cout << std::endl;

    return 0;
}